server/database/add-ratings.js

```diff
@@ -0,0 +1,24 @@
+const dbClient = require('./db_connection.js');
+
+const { insertRating, } = require('./queries.js');
+const { getRatingData, } = require('./../helpers/handle-goals.js');
+
+module.exports = (ratings, dbGoal, finalCallback) => {
+  let count = 0;
+  let errCount = 0;
+
+  ratings.forEach(rating => {
+    dbClient.query(insertRating, getRatingData(rating, dbGoal), (createRatingErr) => {
+      if (createRatingErr) {
+        console.log(createRatingErr);
+        errCount += 1;
+      } else {
+        count += 1;
+      }
+      if (errCount + count === ratings.length) {
+        if (errCount > 0) return finalCallback('failed to add some ratings to database');
```

Comment:
thank you for this! ✨
it looks much nicer, but I managed to make a bit of a mess of it and in the end i've relented and installed async, which i figured might come in useful elsewhere too...